src/server/pps/server/api_server.go

```diff
@@ -3037,6 +3046,11 @@ func (a *apiServer) listPipeline(ctx context.Context, request *pps.ListPipelineR
 				return err
 			}
 		}
+		defer func() { seen++ }()
+		if offset > seen {
+			return nil
+		}
+		emmitted++
```

Comment:
`seen` and `emitted` are confusing here, particularly the `defer func() { seen++ }()`.  I think this would be clearer were it `seen++; if offset >= seen { … }` (I _believe_ that’s correct, without and off-by-one error).